include/zephyr/drivers/fuel_gauge.h

```diff
@@ -222,7 +222,7 @@ struct sbs_gauge_device_chemistry {
  * See fuel_gauge_get_property() for argument description
  */
 typedef int (*fuel_gauge_get_property_t)(const struct device *dev,
-					 struct fuel_gauge_property *props, size_t props_len);
+					 struct fuel_gauge_property *prop);
```

Comment:
Cool but I think this now should take `(dev, prop_type, &prop_value)` rather than the composite structure with the status bit that this does not need anyway now that it has a return code. This way getting a property become much more compact as you can just put the type in the call itself and use an uninitialized value as argument.

Reply:
Oh I was going to do that in a follow-up PR after removing the exit code. Want me to separate out the property type out in in this PR?

Reply:
your choice but I think that our downstreaming person would appreciate to get the api changes in a single pr

Reply:
Then I'll do it in the same PR, along with the exit code. I'll keep things up as separate commits so we can change them to separate PRs easily if wanted.